### PR TITLE
docs(server): config is lost when reply.call not found() is called

### PR DESCRIPTION
**Pull request by cesarvspr**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
Lgtm

File: docs/Reference/Server.md [hunk 1]
```diff
@@ -1426,6 +1426,13 @@ plugins are registered. If you would like to augment the behavior of the default
 arguments `fastify.setNotFoundHandler()` within the context of these registered
 plugins.
 
+> Note: Be aware that some config properties from the request object will be 
+> undefined inside the custom not found handler. E.g:
+> request.routerPath, routerMethod and context.config. 
+> This method design goal is to allow calling the common not found route. If
+> you want to return a per-route customized 404 response, you can just do it in
+> the response itself.
```
Comment (Fdawgs) [inline docs/Reference/Server.md:1434]:
```suggestion
> Note: Some config properties from the request object will be 
> undefined inside the custom not found handler. E.g:
> `request.routerPath`, `routerMethod` and `context.config`. 
> This method design goal is to allow calling the common not found route.
> To return a per-route customized 404 response, you can do it in
> the response itself.
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: docs/Reference/Server.md (hunk 1 above)
Comment (cesarvspr) [reply docs/Reference/Server.md]: thanks for your suggestion, accepted!

**Comment by cesarvspr** [comment]
I see that npm run test  has 2 tests breaking, is there any issue open for this? @Fdawgs @mcollina

**Review by Fdawgs — APPROVED** [review]

File: docs/Reference/Server.md [hunk 2]
```diff
@@ -1426,6 +1426,13 @@ plugins are registered. If you would like to augment the behavior of the default
 arguments `fastify.setNotFoundHandler()` within the context of these registered
 plugins.
 
+> Note: Some config properties from the request object will be 
+> undefined inside the custom not found handler. E.g:
+> `request.routerPath`, `routerMethod` and `context.config`. 
+> This method design goal is to allow calling the common not found route.
+> To return a per-route customized 404 response, you can do it in
+> the response itself.
```
Comment (metcoder95) [inline docs/Reference/Server.md:1434]: This also applies to the newest APIs `Request#routeSchema` and `Request#routeConfig`. 

Ref: https://www.fastify.io/docs/latest/Reference/Request/

**Pull request merged**